wafflemarket/user/views.py

```diff
@@ -50,7 +67,52 @@ def post(self, request, *args, **kwargs):
         except IntegrityError:
             return Response(login(request.data), status=status.HTTP_200_OK)
         return Response(login(request.data), status=status.HTTP_201_CREATED)
-    
+
+class GoogleSigninCallBackApi(APIView):
+    permission_classes = (permissions.AllowAny, )
+
+    def get(self, request, *args, **kwargs):
+        if settings.DEBUG:
+            return render(request, 'user/logintest.html')
+        else:
+            return HttpResponse(status=404)
+
+    def post(self, request, *args, **kwargs):
+        token = request.data.get('token')
+        
+        # verifying access_token
+        try:
+            idinfo = id_token.verify_oauth2_token(token, Request(), os.getenv('GOOGLE_CLIENT_ID'))
+            userid = idinfo['sub']
+        except ValueError:
+            pass
+
+        # request google to get user info
+        user_info_response = requests.get(
+            "https://oauth2.googleapis.com/tokeninfo",
```

Comment:
[tokeninfo 엔드포인트는 디버깅용](https://developers.google.com/identity/sign-in/web/backend-auth#calling-the-tokeninfo-endpoint)이며, 프로덕션 단계에서는 사용하면 안됩니다.

An easy way to validate an ID token signature for debugging is to use the tokeninfo endpoint. Calling this endpoint involves an additional network request that does most of the validation for you while you test proper validation and payload extraction in your own code. It is not suitable for use in production code as requests may be throttled or otherwise subject to intermittent errors.

Reply:
85번째 줄의 idinfo에 필요한 모든 정보가 담겨 있으며, 따로 tokeninfo 엔드포인트를 호출할 필요는 없을 것 같습니다.